ic-os/setupos/config/config.ini

```diff
@@ -23,6 +13,26 @@ ipv6_prefix=2a00:fb01:400:44
 # Define the gateway address for your IPv6 network. Ensure it's within your network range:
 ipv6_gateway=2a00:fb01:400:44::1
 
+# ------------------------------
+# Node Reward Settings (required for some node operators)
+# ------------------------------
+# Specifies the type of node rewards that the node operator expects to receive.
+# The value must match a community-approved value in the NNS.
+#
+# Example command to get node operator details:
+# ic-admin --nns-url https://ic0.app get-node-operator ut325-qbq5v-fli2f-e2a5h-qapdd-fsuyv-xej2j-ogvux-i3fc2-5nj3a-2ae
+# Example output:
+# NodeOperator { node_operator_principal_id: ut325-qbq5v-fli2f-e2a5h-qapdd-fsuyv-xej2j-ogvux-i3fc2-5nj3a-2ae, node_allowance: 0, node_provider_principal_id: lq5ra-f4ibl-t7wpy-hennc-m4eb7-tnfxe-eorgd-onpsl-wervo-7chjj-6qe, dc_id: "bo1", rewardable_nodes: {"type1": 28}, ipv6: None }
+# ==> The community-approved value of node_reward_type for this operator is "type1"
+#
+# If not provided, the NNS will attempt to automatically assign a value based on the operator's reward configuration.
+# If the reward configuration in unique, the NNS will automatically assign the correct node_reward_type.
+# If the reward configuration is ambiguous (e.g., the operator has both "type3" and "type3.1" rewards),
+# node registration with a node_reward_type set to empty value below will fail.
+# Expected value format:
+# node_reward_type=typeX or node_reward_type=typeX.Y
+# Examples value include: type3.1, type1, type1.1, etc.
+# node_reward_type=type3.1
```

Comment:
Now that there's so much text here, it makes the exact line to update less clear:
`# node_reward_type=type3.1`

Maybe some of this information can be moved to the deployment guides?
https://wiki.internetcomputer.org/wiki/Node_Deployment_Guide_(with_an_HSM)
https://wiki.internetcomputer.org/wiki/Node_Deployment_Guide

Reply:
I've tried to separate the line, to make it stand out. I've just had a NP ask me questions about this value so the previous text (on master) was certainly not clear enough.
Not sure how to make it more concise and clear at the same time. We could omit some information here, but not sure if that would improve clarity.

Reply:
I think essentially all the node_reward_type information can be in the deployment guide, and then we can leave the config.ini file clean and concise.

```
# ------------------------------
# Node Reward Settings (required for some node operators)
# ------------------------------
# Specifies the type of node rewards that the node operator expects to receive (e.g.: type3.1, type1, type1.1, etc).
# node_reward_type=type3.1
```

That's what we do for [ipv6 addresses](https://wiki.internetcomputer.org/wiki/Node_Deployment_Guide#B._Edit_Config.ini:~:text=The%20IPv6%20prefix,explicitly%20written%20out.) and [ipv4 addresses](https://wiki.internetcomputer.org/wiki/Node_Deployment_Guide#B._Edit_Config.ini:~:text=Configuring%20your%20node,here%20for%20details.)

Reply:
Several node providers asked me what this value should be and whether this line should be `node_reward_type=1` or similar, although you added the explanation into the deployment guide. No one reads external documents. It's easier to just ask.
The config.ini comment on master [explicitly provides a valid example](https://sourcegraph.com/github.com/dfinity/ic/-/blob/ic-os/setupos/config/config.ini?L12).
So if you don't mind I'd try to add a more detailed comment directly into the `config.ini` and see if people will read then.
If you'd prefer leaving config.ini very short (which seems to be the case?), we can try adding a link to the doc with details and see if that works?
Maybe it's just that they don't know where to find detailed explanation so they fall back to asking.

Reply:
I like the idea of linking to the docs! That would be my preferences, but if you'd rather add a detailed explanation directly in the config.ini, I'm fine with that too 🙂

Reply:
Let's try it with the external docs.